src/charm.py

```diff
@@ -103,67 +145,74 @@ def build_pod_spec(self):
         config_file_contents = self.build_config_file()
 
         # set image details based on what is defined in the charm configuation
-        image_details = {
-            'imagePath': config['alertmanager_image_path']
-        }
+        image_details = {"imagePath": config["alertmanager_image_path"]}
+
+        ha_targets = []
+        num_units = self.num_units()
+        if num_units > 1:
+            for i in range(num_units):
+                cluster_address = UNIT_ADDRESS.format(
+                    self.meta.name, i, self.meta.name, self.model.name
+                )
+                ha_targets.append(f"--cluster.peer={cluster_address}:{HA_PORT}")
 
         spec = {
-            'version': 3,
-            'containers': [{
-                'name': self.app.name,  # self.app.name is defined in metadata.yaml
-                'imageDetails': image_details,
-                'args': [
-                    '--config.file=/etc/alertmanager/alertmanager.yaml',
-                    '--storage.path=/alertmanager',
-                ],
-                'ports': [{
-                    'containerPort': config['port'],
-                    'protocol': 'TCP'
-                }],
-                'kubernetes': {
-                    'readinessProbe': {
-                        'httpGet': {
-                            'path': '/-/ready',
-                            'port': config['port']
+            "version": 3,
+            "containers": [
+                {
+                    "name": self.app.name,  # self.app.name is defined in metadata.yaml
+                    "imageDetails": image_details,
+                    "args": [
+                        "--config.file=/etc/alertmanager/alertmanager.yaml",
+                        "--storage.path=/alertmanager",
+                        f'--web.listen-address=:{config["port"]}',
+                    ]
+                    + ha_targets,
```

Comment:
This implies every time a new alertmanager unit is added, every alertmanager pod is restarted. Is this correct ? Does this mean alertmanager does not provide an API that allows dynamic configuation without pod restart ?